### PR TITLE
common/build-helper/rust.sh: dynamic linking to libpcre2 on musl

### PR DESCRIPTION
**Pull request by Logarithmus**
By default, `pcre2-sys` crate use static linking on `musl` targets. This patch fixes such behavior and all Rust crates will automatically use system-provided `libpcre2` on musl.

### CODE REVIEW

File: common/build-helper/rust.sh [hunk 1]
```diff
@@ -38,3 +38,6 @@ export SODIUM_INC_DIR="${XBPS_CROSS_BASE}/usr/lib"
 
 # openssl-sys
 export OPENSSL_NO_VENDOR=1
+
+# pcre2-sys
```
Comment (ericonr) [inline common/build-helper/rust.sh:42]: Make a note that this is only necessary for musl

**Review by ericonr — COMMENTED** [review]
If you want to include a `ripgrep` revbump in this PR, that should be fine.

**Comment by ericonr** [comment]
@q66 do you know of any issues this could cause?

**Comment by Logarithmus** [comment]
> @q66 do you know of any issues this could cause?

Ping!

**Comment by ericonr** [comment]
@sgn also asked to wait on merging this one.

**Comment by sgn** [comment]
> @sgn also asked to wait on merging this one.

The check for `pcre2` also fail for musl, so, I don't have any objection for this change.

**Comment by Logarithmus** [comment]
> > @sgn also asked to wait on merging this one.
> 
> The check for `pcre2` also fail for musl, so, I don't have any objection for this change.
What "check for `pcre2`" are you talking about?

**Comment by ericonr** [comment]
It's probably the checks being run with `xbps-src -Q pcre2`, at a guess.

**Pull request merged**